mvlearn/embed/kcca.py

```diff
@@ -167,266 +218,60 @@ def fit_transform(self, Xs):
             The data for kcca to fit to.
             Each sample will receive its own embedding.
 
-        Returns
-        -------
         weights_ : list of array-likes
                    Canonical weights
-        components_ : list of array-likes
-                     Canonical components
-        cancorrs_ : list of array-likes
-                   Correlations of the canonical components on
-                   the training dataset
-        """
-        Xs = check_Xs(Xs, multiview=True)
-
-        components_ = kcca(
-            Xs,
-            self.reg,
-            self.n_components,
-            ktype=self.ktype,
-            sigma=self.sigma,
-            degree=self.degree,
-        )
-
-        self.weights_ = _listdot(Xs, components_)
-        self.components_ = _listdot([d.T for d in Xs], self.weights_)
-        self.cancorrs_ = _listcorr(self.components_)
-
-        if len(Xs) == 2:
-            self.cancorrs_ = self.cancorrs_[np.nonzero(self.cancorrs_)]
-        return self
-
-    def validate(self, vdata):
-        """
-        Uses the kCCA mapping and generalizes to other data
-        For each dimension in the test data, correlations between
-        predicted and actual data are computed.
-
-        Parameters
-        ----------
-        vdata: float
-               Standardized data (z-score)
 
         Returns
         -------
-        preds_: list of array-likes
-                Prediction components of test dataset
-        vcorrs_: list of array-likes
-                Correlations on the test dataset
+        self : returns an instance of self
         """
```

Comment:
I think these are backwards. If this inherits from BaseEmbed, then fit() should return self, and transform should return Xs_transformed. https://github.com/neurodata/mvlearn/blob/master/mvlearn/embed/base.py

That way, fit_transform can just call .fit().transform() and return Xs_transformed